packages/middleware-sentry/src/middleware.ts

```diff
@@ -0,0 +1,194 @@
+import * as Sentry from "@sentry/core";
+import { type Span } from "@sentry/types";
+import {
+  InngestMiddleware,
+  type MiddlewareRegisterFn,
+  type MiddlewareRegisterReturn,
+} from "inngest";
+
+/**
+ * Options used to configure the Sentry middleware.
+ */
+export interface SentryMiddlewareOptions {
+  /**
+   * If `true`, the Sentry middleware will not automatically flush events after
+   * each function run. This can be useful if you want to control when events
+   * are sent to Sentry, or leave it to Sentry's default behavior.
+   *
+   * By default, automatic flushing is enabled to ensure that events are sent in
+   * serverless environments where the runtime may be terminated if the function
+   * has returned a value.
+   *
+   * @default false
+   */
+  disableAutomaticFlush?: boolean;
+}
+
+/**
+ * Captures errors and performance data from Inngest functions and sends them to
+ * Sentry.
+ *
+ * Use the `sentryMiddleware()` helper to create a new Sentry middleware.
+ *
+ * This type is used an explicit return type for the `sentryMiddleware` function
+ * to allow better JSR publishing.
+ */
+export type SentryMiddleware = InngestMiddleware<{
+  name: string;
+  init: (...args: Parameters<MiddlewareRegisterFn>) => {
+    onFunctionRun: (
+      ...args: Parameters<
+        NonNullable<MiddlewareRegisterReturn["onFunctionRun"]>
+      >
+    ) => {
+      transformInput: () => {
+        ctx: {
+          /**
+           * The Sentry client fetched by `import * as Sentry from "@sentry/node"`.
+           */
+          sentry: typeof Sentry;
+        };
+      };
+    };
+  };
+}>;
+
+/**
+ * Captures errors and performance data from Inngest functions and sends them to
+ * Sentry.
+ *
+ * This imports Sentry directly and relies on it already being initialized using
+ * `Sentry.init()`. For more information on how to configure Sentry, see the
+ * [Sentry documentation](https://docs.sentry.io/platforms/node/).
+ */
+export const sentryMiddleware = (
+  /**
+   * Options used to configure the Sentry middleware.
+   */
+  opts?: SentryMiddlewareOptions
+): SentryMiddleware => {
+  const mw = new InngestMiddleware({
+    name: "@inngest/middleware-sentry",
+    init({ client }) {
+      return {
+        onFunctionRun({ ctx, fn, steps }) {
+          return Sentry.withScope((scope) => {
+            const sharedTags: Record<string, string> = {
+              "inngest.client.id": client.id,
+              "inngest.function.id": fn.id(client.id),
+              "inngest.function.name": fn.name,
+              "inngest.event": ctx.event.name,
```

Comment:
```suggestion
              "inngest.event.name": ctx.event.name,
```

In case more event fields are added (like `event.id`)

Reply:
Thank you!